File: src/Julia/Take-a-Ten-Minutes-Walk.jl


"""
valid walk 的标准为：走十步并回到原点
"""
function isvalidwalk(walk)
  [length(walk) == 10,
    sum(walk .== 'n') == sum(walk .== 's'),
    sum(walk .== 'e') == sum(walk .== 'w')] |> all
end


using Test
@test isvalidwalk(['n', 's', 'n', 's', 'n', 's', 'n', 's', 'n', 's']) == true

File: src/R/Take-a-Ten-Minutes-Walk.R
#' @description 10轮后恰好回到起点返回 TRUE，否则返回 FALSE
isValidWalk <- function(walk) {
  all(
    length(walk) == 10,
    sum(walk == "n") == sum(walk == "s"),
    sum(walk == "w") == sum(walk == "e")
  )
}
